lib/rubocop/cop/betterment/boolean_query_parameter.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Betterment
+      class BooleanQueryParameter < Base
+        MSG = <<~MSG
+          A boolean fetched from query params will never be false when
+          explicitly specified on the request. Please use
+          ActiveModel::Type::Boolean.new.cast to cast the value to a boolean.
+        MSG
+
+        # @!method fetch_boolean?(node)
+        def_node_matcher :fetch_boolean?, <<-PATTERN
+          (send
+            (send ...) :fetch
+            (sym ...)
+            (boolean))
+        PATTERN
+
+        # @!method boolean_cast?(node)
+        def_node_search :boolean_cast?, <<-PATTERN
+          (send
+            (send
+              (const
+                (const
+                  (const nil? :ActiveModel) :Type) :Boolean) :new) :cast
+            ...)
+        PATTERN
+
+        # @!method action_controller?(node)
+        def_node_search :action_controller?, <<~PATTERN
+          {
+            (const {nil? cbase} :ApplicationController)
+            (const (const {nil? cbase} :ActionController) :Base)
+          }
+        PATTERN
+
+        def initialize(config = nil, options = nil)
+          super(config, options)
+        end
+
+        def on_send(node)
+          return unless fetch_boolean?(node)
+
+          return unless inherit_action_controller_base?(node)
+
+          return if node.each_ancestor(:send).any? { |ancestor| boolean_cast?(ancestor) }
+
+          add_offense(node)
+        end
+
+        private
+
+        def inherit_action_controller_base?(node)
+          class_node = node.each_ancestor(:class).first
```

Comment:
__question:__ I assume if there are multiple class ancestors, this will return the one that is most closely wrapping the node?

Reply:
That's a good question. I borrowed this approach from this cop: https://github.com/rubocop/rubocop-rails/blob/master/lib/rubocop/cop/rails/action_controller_flash_before_render.rb#L106-L108

I could add a test case where we have a nested class. In that instance, we probably do want to match where _any_ containing class is a Controller? The only reason to limit it to controllers is because that is a good heuristic that we are probably dealing with query params rather than some other Hash

Reply:
> I could add a test case where we have a nested class.

I'd be ok with that. Not sure if we have mutant set up on this repo, but I'd typically write at least one test for each of the possible states.